From 6d7e523775e696ea489f993425bf7772941dfd9b Mon Sep 17 00:00:00 2001
From: Jian He <jianhe@apache.org>
Date: Wed, 7 Jan 2015 14:12:07 -0800
Subject: [PATCH 1138/2848] YARN-3010. Fixed findbugs warning in
 AbstractYarnScheduler. Contributed by Yi Liu
 (cherry picked from commit
 e13a484a2be64fb781c5eca5ae7056cbe194ac5e) (cherry
 picked from commit
 d3b26454e453953d2eb221b62d1ff3632da9e8a6)

Conflicts:
	hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/fair/FairScheduler.java

Change-Id: Ib8bf84d59e38f9b0168f31ee28104380accd1e70
---
 .../scheduler/AbstractYarnScheduler.java           |    4 ++--
 .../scheduler/capacity/CapacityScheduler.java      |    3 ++-
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/AbstractYarnScheduler.java b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/AbstractYarnScheduler.java
index bc206db..753259c 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/AbstractYarnScheduler.java
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/AbstractYarnScheduler.java
@@ -192,8 +192,8 @@ protected void initMaximumResourceCapability(Resource maximumAllocation) {
     }
   }
 
-  protected void containerLaunchedOnNode(ContainerId containerId,
-                                         SchedulerNode node) {
+  protected synchronized void containerLaunchedOnNode(
+      ContainerId containerId, SchedulerNode node) {
     // Get the application for the finished container
     SchedulerApplicationAttempt application = getCurrentAttemptForContainer
         (containerId);
diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/capacity/CapacityScheduler.java b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/capacity/CapacityScheduler.java
index 2edce2b..5691d91 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/capacity/CapacityScheduler.java
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/capacity/CapacityScheduler.java
@@ -245,7 +245,8 @@ public CapacitySchedulerConfiguration getConfiguration() {
   }
 
   @Override
-  public RMContainerTokenSecretManager getContainerTokenSecretManager() {
+  public synchronized RMContainerTokenSecretManager 
+  getContainerTokenSecretManager() {
     return this.rmContext.getContainerTokenSecretManager();
   }
 
-- 
1.7.9.5

